sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -198,7 +198,8 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
 
         // Select the result of the first aggregate in the last aggregate.
         val result = AggregateExpression(
-          aggregate.First(evalWithinGroup(regularGroupId, operator.toAttribute), Literal(true)),
+          aggregate.First("first",
+            evalWithinGroup(regularGroupId, operator.toAttribute), Literal(true)),
```

Comment:
a bit verbose to always set "first" for `First`, so how about defining auxiliary constructor?
```
  def this(child: Expression, ignoreNullsExpr: Expression) =
    this("first", child, Literal.create(false, BooleanType))
```

Reply:
Done